packages/mongodb-ns/src/index.ts

```diff
@@ -0,0 +1,142 @@
+type NS = {
+  ns: string;
+  dotIndex: number;
+  database: string;
+  collection: string;
+  system: boolean;
+  isSystem(): boolean;
+  oplog: boolean;
+  isOplog(): boolean;
+  command: boolean;
+  isCommand(): boolean;
+  special: boolean;
+  isSpecial(): boolean;
+  specialish: boolean;
+  normal: boolean;
+  isNormal(): boolean;
+  validDatabaseName: boolean;
+  validCollectionName: boolean;
+  databaseHash: number;
+  toString(): string;
+  // Assigned in the constructor, but will always be undefined
+  isConf(): undefined;
+};
+
+type NSConstructor = {
+  new (ns: string | NS): NS;
```

Comment:
```suggestion
type NSConstructor = {
  (ns: string | NS): NS;
  new (ns: string | NS): NS;
```

Sorry!! one more thing, since our goal is to NOT change all the call sites of this.

since compass almost universally imports like:
```ts
import toNS from 'mongodb-ns';
// and calls it like
toNS('string');
```

If we don't add the plain `(ns: string | NS): NS;` in addition to the `new` one you'll be required to add `new` everywhere by TS.